packages/babel-core/src/config/index.js

```diff
@@ -161,6 +160,30 @@ export default function loadConfig(inputOpts: mixed): ResolvedConfig | null {
   };
 }
 
+function merge(target: ValidatedOptions, source: ValidatedOptions): void {
+  for (const k of Object.keys(source)) {
+    if (k === "parserOpts" && source.parserOpts) {
+      const parserOpts = source.parserOpts;
+      const targetObj = (target.parserOpts = target.parserOpts || {});
+      mergeObject(targetObj, parserOpts);
+    } else if (k === "generatorOpts" && source.generatorOpts) {
+      const generatorOpts = source.generatorOpts;
+      const targetObj = (target.generatorOpts = target.generatorOpts || {});
+      mergeObject(targetObj, generatorOpts);
+    } else {
+      const val = source[k];
+      if (val !== undefined) target[k] = (val: any);
+    }
+  }
+}
+
+function mergeObject<T: {}>(target: T, source: T) {
```

Comment:
maybe we could figure out some better name for this one? i had to look into the implementation to see how it is different from `Object.assign`, maybe just smth like `safeMerge`?

Reply:
How about `mergeDefaults` since that seems to be the common name for this operation?

Reply:
Sounds good to me 😃